CoordImporter/Plugin.cs

```diff
@@ -25,12 +14,13 @@ public sealed class Plugin : IDalamudPlugin
         public string Name => "Coordinate Importer";
         private const string CommandName = "/ci";
 
-        public static DalamudPluginInterface PluginInterface { get; private set; }
+        public static DalamudPluginInterface PluginInterface { get; set; } = null!;
+        public static IChatGui Chat { get; set; } = null!;
+        public static IPluginLog Logger { get; set; } = null!;
+        public static IDataManagerManager DataManagerManager { get; set; } = null!;
```

Comment:
i had to give these public setters so they could be mocked in the unit tests. i'm very open to us considering changing from public static access to proper dependency injection, though. i actually only went with the static access originally because i saw it in some core dalamud code, and thought it was a nice pattern for convenience, even though it's technically Wrong™.

Reply:
I haven't looked too deep into Dalamud code but I suspect that approach is probably being phased out since they are (at least for plugin-facing stuff) going towards interface-all-the-things, meaning we have to do DI anyway. So we might as well just do DI everywhere 🤷

Reply:
oh i misread at first. i'm actually not sure what you mean, now :thinky:. we can pull in all these services using di, in the plugin constructor, but we can still expose them as public static, so they are easier to access in other classes. like, there's a convenience to doing `Plugin.Logger` rather than having to pass a logger to every object we instantiate that uses one. it's slightly worth the annoyance of doing that, to make testing mildly easier and code more maintainable, but i just haven't cared enough about the benefits. but i wholly support it!

in all my changes, and in scout helper, i've basically been doing public static access for core things like `PluginInterface`, `Logger`, `Chat`, and `DataManager`, and passing everything else in as constructor parameters. but it's certainly a somewhat arbitrary distinction :P. and as i said, i'm very on board to have everything do di ('w')b